lib/topological_inventory/sync/host_inventory_response_worker.rb

```diff
@@ -0,0 +1,160 @@
+require "concurrent"
+require 'stringio'
+
+module TopologicalInventory
+  class Sync
+    class ResponseWorker
+      attr_reader :started
+      alias started? started
+
+      def initialize(config, logger)
+        self.config              = config
+        self.lock                = Mutex.new
+        self.timeout_lock        = Mutex.new
+        self.logger              = logger
+        self.started             = Concurrent::AtomicBoolean.new(false)
+        self.registered_messages = Concurrent::Map.new
+        self.workers             = {}
+      end
+
+      # Start listening on Kafka
+      def start
+        lock.synchronize do
+          return if started.value
+
+          started.value = true
+          workers[:maintenance] = Thread.new { check_timeouts while started.value }
+          workers[:listener]    = Thread.new { listen while started.value }
+        end
+      end
+
+      # Stop listener
+      def stop
+        lock.synchronize do
+          return unless started.value
+
+          started.value = false
+          workers[:listener]&.terminate
+          workers[:maintenance]&.join
+        end
+      end
+
+      def register_message(external_id, source_id)
+        registered_messages[external_id] = {
+          :source_id  => source_id,
+          :created_at => Time.now.utc
+        }
+      end
+
+      private
+
+      attr_accessor :config, :lock, :timeout_lock, :logger, :workers, :registered_messages
+      attr_writer :started
+
+      def listen
+        # Open a connection to the messaging service
+        client = ManageIQ::Messaging::Client.open(default_messaging_opts)
+
+        logger.info("Topological Inventory Sync Response worker started...")
+        client.subscribe_topic(queue_opts) do |message|
+          process_message(message)
+        end
+      rescue => err
+        logger.error("Exception in kafka listener: #{err}\n#{err.backtrace.join("\n")}")
+      ensure
+        client&.close
+      end
+
+      def process_message(message)
+        response       = JSON.parse(message.payload)
+        host           = response['host'] if response
+        external_id    = host['external_id'] if host
+        registered_msg = registered_messages[external_id] if external_id
+        source_id      = registered_msg[:source_id] if registered_msg
+
+        return unless response['type'] == 'created'
```

Comment:
I think we should move this to 70 - don't want to be populating `registered_messages` with messages we don't really care about (could be a memory leak for long-running pods).